paimon-core/src/main/java/org/apache/paimon/table/FileStoreTableFactory.java

```diff
@@ -38,12 +37,8 @@
 public class FileStoreTableFactory {
 
     public static FileStoreTable create(CatalogContext context) {
-        FileIO fileIO;
-        try {
-            fileIO = FileIO.get(CoreOptions.path(context.options()), context);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
+        FileIO fileIO = new ResolvingFileIO();
```

Comment:
Do we need to modify the default value?

Reply:
I think we need, if user use external path, then the original fileio can not work.

Reply:
I think you should enable it through catalog config.

Reply:
I started to think the same way. In this case, simply adding the table attribute `data-file.external-paths` by the user cannot achieve the purpose of writing data to the external path, and users also need to modify the attributes of the catalog. For example, `enable_external_path`. Do you think this usage is okay?

Reply:
I think it could be more direct, add catalog option: resolving-fileio.enabled

Reply:
done